examples/metrics/application_efficiency.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: 0BSD
+"""
+Working with Application Efficiency
+===================================
+
+Understanding relative performance.
+
+One goal of P3 analysis is to understand how well a given application is able
+to adapt to and make effective use of the capabilities of different
+platforms. Comparisons of *raw* performance (e.g., time to solution) across
+platforms can't help us, because raw performance doesn't reflect how fast an
+application **should** run.
+
+To address this, the P3 Analysis Library works with normalized *performance
+efficiency* data. With normalized data, an application's performance can be
+represented as a number in the range :math:`[0, 1]`, where :math:`1` means
+an application is achieving the best possible performance.
```

Comment:
Just a better segue into the next section: 

> There are multiple ways we can normalize the data to measure relative efficiency, and for this tutorial we will consider _application efficiency_.

Reply:
With this added segue I think it reads quite strangely now. "... we will consider application efficiency... the simplest we can work with is application efficiency, defined below:"

I would perhaps suggest then dropping the "Application Efficiency" title, and joining the sentences either side to:

> There are multiple ways we can normalise the data to measure relative efficiency, and for this tutorial we will consider *application efficiency*, defined as: ...

Reply:
Good point.  I've made this change in 615732c8cc7a693e0ce0d269e734a05bd2ad95fa.